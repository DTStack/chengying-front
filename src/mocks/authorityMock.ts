const authority = [
    'menu_cluster_manage',
    'sub_menu_cluster_manage',
    'cluster_view',
    'cluster_edit',
    'sub_menu_cluster_overview',
    'sub_menu_cluster_host',
    'sub_menu_cluster_image_store',
    'image_store_view',
    'image_store_edit',
    'menu_app_manage',
    'sub_menu_package_manage',
    'package_view',
    'package_upload_delete',
    'sub_menu_installed_app_manage',
    'installed_app_view',
    'menu_deploy_guide',
    'menu_product_overview',
    'menu_service',
    'service_view',
    'service_product_start_stop',
    'service_start_stop',
    'service_roll_restart',
    'service_config_edit',
    'service_config_distribute',
    'service_dashboard_view',
    'menu_product_host',
    'menu_product_diagnosis',
    'sub_menu_log_view',
    'log_view',
    'log_download',
    'sub_menu_event_diagnosis',
    'sub_menu_config_change',
    'menu_monitor',
    'sub_menu_dashboard',
    'sub_menu_alarm',
    'sub_menu_alarm_record',
    'alarm_record_view',
    'alarm_record_open_close',
    'sub_menu_alarm_channel',
    'alarm_channel_view',
    'alarm_channel_edit',
    'menu_user_manage',
    'sub_menu_user_manage',
    'user_view',
    'user_add',
    'user_edit',
    'user_delete',
    'user_able_disable',
    'user_reset_password',
    'sub_menu_role_manage',
    'sub_menu_user_info',
    'menu_security_audit'
];

function getAuthorityList() {
    const obj = {};
    authority.forEach((item: string) => {
        obj[item] = true;
    })
    return obj;
}

export const authorityList = getAuthorityList();
