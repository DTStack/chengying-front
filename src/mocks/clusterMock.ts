export const cur_parent_cluster = {
    id: 46,
    type: 'kubernetes',
    name: 'DTLogger',
    mode: 0
}
export default {

}
