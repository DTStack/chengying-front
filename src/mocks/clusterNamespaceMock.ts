export default {
    getNamespaceList: {
        msg: 'ok',
        code: 0,
        data: [
            {
                id: 26,
                namespace: 'dt-test',
                status: 'valid',
                cpu_used: '1.5core',
                cpu_total: '300core',
                cpu_percent: 0.5,
                memory_used: '2.48GB',
                memory_total: '300GB',
                mem_percent: 0.8294270833333334,
                user: 'admin@dtstack.com',
                update_time: '2020-12-18 17:43:59'
            }
        ]
    }
}
