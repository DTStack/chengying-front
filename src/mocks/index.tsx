export { default as clusterMnagerMock } from './clusterMnagerMock';
export { default as ServiceMock } from './serviceMock';
export { default as ImageStoreMock, imageStoreInfo } from './imageStoreMock';
export { default as ServicePageMock } from './servicePageMock';
export { default as DeployMock } from './deployMock';
export { default as DashBoardMock } from './dashboardMock';
export { default as UserCenterMock } from './userCenterMock';
export { default as SecurityAuditMock } from './securityAuditMock';
export { default as ClusterNamespaceMock } from './clusterNamespaceMock';
export { cur_parent_cluster } from './clusterMock';
export { authorityList } from './authorityMock';
