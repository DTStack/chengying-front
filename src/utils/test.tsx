import * as React from 'react';
import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunkMiddleware from 'redux-thunk';

// this is a handy function that I normally make available for all my tests
// that deal with connected components.
// you can provide initialState for the entire store that the ui is rendered with
export function renderWithRedux(
  Component,
  rootReducer,
  initialState: any = {}
) {
  const store: any = createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunkMiddleware)
  );
  const layout = <Provider store={store}>{Component}</Provider>;
  return {
    ...render(layout),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}
