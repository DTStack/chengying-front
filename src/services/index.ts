import apis from '@/constants/apis';
import * as http from '@/utils/http';

const { product, service, instance, host, alert, cluster, common } = apis;

export const Service = {
  addNetworkPackage(params: any) {
    return http[product.addNetworkPackage.method](
      product.addNetworkPackage.url,
      params
    );
  },
  getProductpackageList() {
    return http[product.getProductpackageList.method](
      product.getProductpackageList.url
    );
  },
  delProductPackageItem(params: any) {
    return http[product.delProductpackageItem.method](
      product.delProductpackageItem.url,
      params
    );
  },
  getAllProducts(params: any) {
    return http[product.getProductList.method](
      product.getProductList.url,
      params
    );
  },
  Verifylink(params: any) {
    return http[product.Verifylink.method](product.Verifylink.url, params);
  },
  // 获取更新目录
  getPatchPath(params: any) {
    return http[product.getPatchPath.method](product.getPatchPath.url, params);
  },
  // 上传更新接口
  updatePatchPath(params: any) {
    return http[product.updatePatchPath.method](
      product.updatePatchPath.url,
      params
    );
  },
  // 获取更新状态
  updatePatchStatus(params: any) {
    return http[product.updatePatchStatus.method](
      product.updatePatchStatus.url,
      params
    );
  },
  getParentProductList(params?: any) {
    return http[product.getParentProductList.method](
      product.getParentProductList.url,
      params
    );
  },
  // 获取产品列表
  getClusterProductList(params?: any) {
    return http[product.getClusterProductList.method](
      product.getClusterProductList.url,
      params
    );
  },
  getCurrentProduct(params: any) {
    return http[product.getCurrentProduct(params).method](
      product.getCurrentProduct(params).url,
      params
    );
  },
  getProductUpdateRecords(params: any) {
    return http[product.getProductUpdateRecords(params).method](
      product.getProductUpdateRecords(params).url,
      params
    );
  },
  // 获取补丁包更新历史
  getProductUpdate(params: any) {
    return http[product.getProductUpdate(params).method](
      product.getProductUpdate(params).url,
      params
    );
  },
  // 获取产品包升级目标版本列表
  getProductVersionList(params: any) {
    const { upgrade_mode } = params;
    return http[product.getProductVersionList(params).method](
      product.getProductVersionList(params).url,
      {
        upgrade_mode,
      }
    );
  },
  getProductDeployRecords(params: any) {
    return http[product.getProductDeployRecords(params).method](
      product.getProductDeployRecords(params).url,
      params
    );
  },
  getServiceDashInfo(params: any) {
    const url = service.getServiceDashInfo.url;
    const { product_name, service_name } = params;
    return http[service.getServiceDashInfo.method](
      `${url}?tag=${product_name}&tag=${service_name}`
    );
  },
  getInstanceLog(params: any) {
    return http[instance.getInstanceLog(params).method](
      instance.getInstanceLog(params).url,
      params
    );
  },
  getFilterData(params: any) {
    return http[host.getFilterData.method](host.getFilterData.url, params);
  },
  // 获取主机集群下的主机列表
  getHostClusterHostList(params: any) {
    return http[common.getHostClusterHostList.method](
      common.getHostClusterHostList.url,
      params
    );
  },
  // 获取kubernetes集群下的主机列表
  getKubernetesClusterHostList(params: any) {
    return http[common.getKubernetesClusterHostList.method](
      common.getKubernetesClusterHostList.url,
      params
    );
  },
  // 获取主机列表
  getClusterHostList(params: any, type: string) {
    return type === 'hosts'
      ? this.getHostClusterHostList(params)
      : this.getKubernetesClusterHostList(params);
  },
  // 获取集群下主机分组列表
  getClusterhostGroupLists(params: any) {
    return http[common.getClusterhostGroupLists.method](
      common.getClusterhostGroupLists.url,
      params
    );
  },
  getTableData(params: any) {
    return http[host.getTableData.method](host.getTableData.url, params);
  },
  showInstallInfo(params: any) {
    return http[host.showInstallInfo.method](host.showInstallInfo.url, params);
  },
  getAlertsHistory(params: any) {
    return http[alert.getAlertsHistory(params).method](
      alert.getAlertsHistory(params).url,
      params
    );
  },
  getHostInstance(params: any) {
    return http[host.getHostInstance.method](host.getHostInstance.url, params);
  },
  getHostServicesList(params: any) {
    return http[host.getHostServicesList.method](
      host.getHostServicesList.url,
      params
    );
  },
  getDeployShot(params: any) {
    return http[product.getDeployShot(params).method](
      product.getDeployShot(params).url
    );
  },
  getPatchUpdateShot(params: any) {
    return http[product.getPatchUpdateShot(params).method](
      product.getPatchUpdateShot(params).url
    );
  },
  delProduct(params: any) {
    return http[product.delProduct(params).method](
      product.delProduct(params).url
    );
  },
  getClusterList(params: any) {
    return http[cluster.getClusterList.method](
      cluster.getClusterList.url,
      params
    );
  },
};

export { default as productService } from './productService';
export { default as dashboardService } from './dashboardService';
export { default as alertRuleService } from './alertRuleService';
export { default as alertChannelService } from './alertChannelService';
export { default as installGuideService } from './installGuideService';
export { default as addHostService } from './addHostService';
export { default as deployService } from './deployService';
export { default as servicePageService } from './ServicePageService';
export { default as createHostService } from './createHostService';
export { default as userCenterService } from './userCenterService';
export { default as eventService } from './eventService';
export { default as configService } from './configService';
export { default as clusterManagerService } from './clusterManagerService';
export { default as clusterHostService } from './clusterHostService';
export { default as imageStoreService } from './imageStoreService';
export { default as SecurityAuditService } from './securityAuditService';
export { default as ClusterNamespaceService } from './clusterNamespaceService';
export { default as inspectionReportService } from './inspectionReportService';
export { default as echoService } from './echoService';
export { default as securityService } from './securityService';
export { default as scriptManager } from './scriptManager';
export { default as globalConfig } from './globalConfig';
export { default as productLine } from './productLine';