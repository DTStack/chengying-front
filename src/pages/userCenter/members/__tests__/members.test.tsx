import * as React from 'react';
import { cleanup, fireEvent, RenderResult } from '@testing-library/react';
import { renderWithRedux } from '@/utils/test';
import Members from '../index';
import reducer from '@/stores';
import { authorityList, UserCenterMock, clusterMnagerMock } from '@/mocks';
import { userCenterService, clusterManagerService } from '@/services';
import '@testing-library/jest-dom/extend-expect';

jest.mock('@/services');

const defaultProps = {
  authorityList,
};
const reqParams = {
  start: 0,
  limit: 10,
  username: '',
  'sort-by': 'create_time',
  'sort-dir': 'desc',
};
describe('members render', () => {
  let wrapper: RenderResult;
  beforeEach(() => {
    (userCenterService.getMembers as any).mockResolvedValue({
      data: UserCenterMock.getMembers,
    });
    (clusterManagerService.getClusterLists as any).mockResolvedValue({
      data: clusterMnagerMock.getClusterList,
    });
    wrapper = renderWithRedux(<Members {...defaultProps} />, reducer, {
      UserCenterStore: { authorityList },
    });
  });

  afterEach(cleanup);

  // 搜索框
  test('search input unit test', () => {
    const searchInput = wrapper.getByPlaceholderText('按账号/姓名搜索');
    expect(searchInput).toBeInTheDocument();
    // 搜索
    const searchName = 'test';
    fireEvent.change(searchInput, { target: { value: searchName } });
    fireEvent.keyDown(searchInput, { keyCode: 13 });
    expect(userCenterService.getMembers).toHaveBeenCalledTimes(2);
    expect(userCenterService.getMembers).toHaveBeenLastCalledWith({
      ...reqParams,
      username: searchName,
    });
  });

  // 创建账号
  test('create member', () => {
    const createBtn = wrapper.getByRole('button');
    expect(createBtn.textContent).toBe('创建账号');
    // click
    fireEvent.click(createBtn);
    const userInput = wrapper.getByPlaceholderText('请输入账号，允许中英文');
    expect(userInput).toBeInTheDocument();
  });

  // 表格
  test('list render', () => {
    const columns = wrapper.container.getElementsByClassName(
      'ant-table-column-title'
    );
    // columns render
    for (const i in columns) {
      const titles = [
        '账号',
        '姓名',
        '邮箱',
        '手机号',
        '账号状态',
        '角色',
        '创建时间',
        '操作',
      ];
      const column = columns[i];
      if (column instanceof HTMLElement) {
        expect(column.textContent).toBe(titles[i]);
      }
    }

    // 角色 & action
    const rows = wrapper.container.getElementsByClassName('ant-table-row');
    const listMock = UserCenterMock.getMembers.data.list;
    for (const i in rows) {
      const row = rows[i];
      if (row instanceof HTMLElement) {
        const tds = row.getElementsByTagName('td');
        // 角色
        const role = tds[5];
        expect(role.textContent).toBe(listMock[i].role_name);
        // 操作 管理员账号 - 仅可重置密码
        const action = tds[tds.length - 1];
        if (listMock[i].role_name === 'Administrator') {
          expect(action.textContent).toBe('重置密码');
        } else {
          // 以 operator 为例
          if (listMock[i].role_name === 'Cluster Operator') {
            testActions(action);
          }
        }
      }
    }

    // 操作 - 以禁用为例
    async function testActions(action: HTMLElement) {
      const actions = action.getElementsByTagName('a');
      const forbiddenBtn = actions[0];
      /* 禁用 */
      expect(forbiddenBtn).toBeInTheDocument();
      expect(forbiddenBtn.textContent).toBe('禁用');
      // 点击出弹框
      fireEvent.click(forbiddenBtn);
      const modalTitle = wrapper.getByText('确定禁用该成员？');
      expect(modalTitle).toBeInTheDocument();
      // 点击确认禁用
      const okBtn = modalTitle.parentElement.nextElementSibling.lastChild;
      (userCenterService.taggleStatus as any).mockResolvedValue({
        data: { code: 0, data: null, msg: 'ok' },
      });
      (userCenterService.getMembers as any).mockResolvedValue({
        data: UserCenterMock.getMembers,
      });
      fireEvent.click(okBtn);
      expect(userCenterService.getMembers).toHaveBeenLastCalledWith(reqParams);
    }
  });

  test('members Snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
