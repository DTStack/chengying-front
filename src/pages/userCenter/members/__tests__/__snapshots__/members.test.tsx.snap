// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`members render members Snapshot 1`] = `
<DocumentFragment>
  <div
    class="members-container"
    style="display: flex;"
  >
    <div
      style="width: 100%; padding: 20px;"
    >
      <div
        class="members-container-header"
      >
        <span
          class="ant-input-search dt-form-shadow-bg mb-12 ant-input-affix-wrapper"
          style="width: 264px;"
        >
          <input
            class="ant-input"
            placeholder="按账号/姓名搜索"
            type="text"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <i
              aria-label="icon: search"
              class="anticon anticon-search ant-input-search-icon"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="search"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </i>
          </span>
        </span>
        <div
          class="header-r"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              创建账号
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-table-wrapper dt-table-fixed-base"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-default ant-table-fixed-header ant-table-scroll-position-left ant-table-layout-fixed"
            >
              <div
                class="ant-table-content"
              >
                <div
                  class="ant-table-scroll"
                >
                  <div
                    class="ant-table-header"
                  >
                    <table
                      class=""
                    >
                      <colgroup>
                        <col
                          style="width: 15%; min-width: 15%;"
                        />
                        <col
                          style="width: 100px; min-width: 100px;"
                        />
                        <col
                          style="width: 15%; min-width: 15%;"
                        />
                        <col />
                        <col />
                        <col />
                        <col
                          style="width: 15%; min-width: 15%;"
                        />
                        <col
                          style="width: 18%; min-width: 18%;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-row-cell-break-word"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  账号
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-row-cell-break-word"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  姓名
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-row-cell-break-word"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  邮箱
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                          </th>
                          <th
                            class=""
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  手机号
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  账号状态
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                            <i
                              aria-label="icon: filter"
                              class="anticon anticon-filter ant-dropdown-trigger"
                              tabindex="-1"
                              title="Filter menu"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="filter"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                />
                              </svg>
                            </i>
                          </th>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  角色
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                            <i
                              aria-label="icon: filter"
                              class="anticon anticon-filter ant-dropdown-trigger"
                              tabindex="-1"
                              title="Filter menu"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="filter"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                />
                              </svg>
                            </i>
                          </th>
                          <th
                            class="ant-table-column-has-actions ant-table-column-has-sorters ant-table-row-cell-break-word"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  创建时间
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                >
                                  <div
                                    class="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                                    title="Sort"
                                  >
                                    <i
                                      aria-label="icon: caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                    <i
                                      aria-label="icon: caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down off"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class=""
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </i>
                                  </div>
                                </span>
                              </div>
                            </span>
                          </th>
                          <th
                            class="ant-table-row-cell-break-word ant-table-row-cell-last"
                          >
                            <span
                              class="ant-table-header-column"
                            >
                              <div>
                                <span
                                  class="ant-table-column-title"
                                >
                                  操作
                                </span>
                                <span
                                  class="ant-table-column-sorter"
                                />
                              </div>
                            </span>
                          </th>
                        </tr>
                      </thead>
                    </table>
                  </div>
                  <div
                    class="ant-table-body"
                    style="overflow-y: scroll;"
                    tabindex="-1"
                  >
                    <table
                      class=""
                    >
                      <colgroup>
                        <col
                          style="width: 15%; min-width: 15%;"
                        />
                        <col
                          style="width: 100px; min-width: 100px;"
                        />
                        <col
                          style="width: 15%; min-width: 15%;"
                        />
                        <col />
                        <col />
                        <col />
                        <col
                          style="width: 15%; min-width: 15%;"
                        />
                        <col
                          style="width: 18%; min-width: 18%;"
                        />
                      </colgroup>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="weixing@dtstack.com"
                        >
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            weixing@dtstack.com
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            <span
                              class="name"
                            >
                              weixing
                            </span>
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            weixing@dtstack.com
                          </td>
                          <td
                            class=""
                          >
                            --
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            启用
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            Cluster Reader
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-sorters ant-table-row-cell-break-word"
                          >
                            2020-09-10 11:48:21
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            <span>
                              <span>
                                <span>
                                  <a>
                                    禁用
                                  </a>
                                  <div
                                    class="ant-divider ant-divider-vertical"
                                    role="separator"
                                  />
                                </span>
                                <span>
                                  <a>
                                    编辑
                                  </a>
                                  <div
                                    class="ant-divider ant-divider-vertical"
                                    role="separator"
                                  />
                                </span>
                                <span>
                                  <a>
                                    删除
                                  </a>
                                  <div
                                    class="ant-divider ant-divider-vertical"
                                    role="separator"
                                  />
                                </span>
                              </span>
                              <a>
                                重置密码
                              </a>
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="yuwan"
                        >
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            yuwan
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            <span
                              class="name"
                            >
                              yuwan
                            </span>
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            yuwan@dtstack.com
                          </td>
                          <td
                            class=""
                          >
                            13512312121
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            启用
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            Cluster Operator
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-sorters ant-table-row-cell-break-word"
                          >
                            2020-09-03 10:15:29
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            <span>
                              <span>
                                <span>
                                  <a>
                                    禁用
                                  </a>
                                  <div
                                    class="ant-divider ant-divider-vertical"
                                    role="separator"
                                  />
                                </span>
                                <span>
                                  <a>
                                    编辑
                                  </a>
                                  <div
                                    class="ant-divider ant-divider-vertical"
                                    role="separator"
                                  />
                                </span>
                                <span>
                                  <a>
                                    删除
                                  </a>
                                  <div
                                    class="ant-divider ant-divider-vertical"
                                    role="separator"
                                  />
                                </span>
                              </span>
                              <a>
                                重置密码
                              </a>
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="admin@dtstack.com"
                        >
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            admin@dtstack.com
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            <span
                              class="name"
                            >
                              admin
                            </span>
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            admin@dtstack.com
                          </td>
                          <td
                            class=""
                          >
                            13345678901
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            启用
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-filters"
                          >
                            Administrator
                          </td>
                          <td
                            class="ant-table-column-has-actions ant-table-column-has-sorters ant-table-row-cell-break-word"
                          >
                            2020-06-03 11:06:56
                          </td>
                          <td
                            class="ant-table-row-cell-break-word"
                          >
                            <span>
                              <a>
                                重置密码
                              </a>
                            </span>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination mini"
              unselectable="unselectable"
            >
              <li
                class="ant-pagination-total-text"
              >
                <span>
                  共
                  <span
                    style="color: rgb(63, 135, 255);"
                  >
                    9
                  </span>
                  条数据，每页显示10条
                </span>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-disabled ant-pagination-prev"
                title="Previous Page"
              >
                <a
                  class="ant-pagination-item-link"
                >
                  <i
                    aria-label="icon: left"
                    class="anticon anticon-left"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </i>
                </a>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a>
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-disabled ant-pagination-next"
                title="Next Page"
              >
                <a
                  class="ant-pagination-item-link"
                >
                  <i
                    aria-label="icon: right"
                    class="anticon anticon-right"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                      />
                    </svg>
                  </i>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
